app/services/ssn_formatter.rb

```diff
@@ -0,0 +1,11 @@
+module SsnFormatter
+  def self.format(ssn, mask: false)
```

Comment:
I think this might be easier as two methods?

```
SsnFormatter.format(ssn)
SsnFormatter.format_masked(ssn)
```

and maybe a 3rd helper method if we wanted to keep that normalization code?

```
SsnFormatter.normalize("123-45-6789")
# => "123456789"
```

Reply:
Yeah, definitely open to revising this. I was mostly following the `PhoneFormatter` as an example. In fact, prior to realizing `PhoneFormatter` existed, I'd implemented this as a helper which looks a lot more like what you propose 😄 

```rb
module SsnHelper
  def format_ssn(ssn)
    normalized_ssn = normalize_ssn(ssn)
    "#{normalized_ssn[0..2]}-#{normalized_ssn[3..4]}-#{normalized_ssn[5..8]}"
  end

  def mask_ssn(ssn)
    first, *, last = normalize_ssn(ssn).chars
    "#{first}**-**-***#{last}"
  end

  private

  def normalize_ssn(ssn)
    ssn.to_s.gsub(/\D/, '')[0..8]
  end
end

```

Reply:
Yeah when we used reek, it flagged [control parameter](https://github.com/troessner/reek/blob/master/docs/Control-Parameter.md) lints and in this particular case, I think it would have been an OK lint to listen to

Anyways, on a scale of 1-10, I think I'm like a 3 or 4 on how strongly I feel about this one

Reply:
> Yeah when we used reek, it flagged [control parameter](https://github.com/troessner/reek/blob/master/docs/Control-Parameter.md) lints and in this particular case, I think it would have been an OK lint to listen to

Thanks for the resource. It's a prototypical example at that! 😅  (save for the common normalization)

I'll plan to update.

Reply:
Updated in 33a72cf.